src/ops/unit.rs

```diff
@@ -0,0 +1,52 @@
+use std::ops::Sub;
+
+pub trait Unit: Sub<Output = Self> + Ord + Copy {
```

Comment:
I think we are doing something similar to the [num-traits](https://crates.io/crates/num-traits) crate here.

I do not have any specific suggestion/wish here, I just want to note that our `Unit` can probably be "better" expressed by one or more traits from `num-traits`. We can look into this at some point.